Cargo.toml

```diff
@@ -31,11 +31,11 @@ thiserror = "1.0"
 [dev-dependencies]
 askama = "0.10"
 
-# This only supports up to 0.7. See examples/actix/main.rs for more.
-actix = "0.9"
-actix-web = { version = "2.0", features = ["openssl"] }
-actix-session = "0.3"
-actix-files = "0.2"
+# Example, use the version numbers you need
+tide = "0.15"
+# tide-rustls = "0.1"
```

Comment:
Not sure what you want to do about this—I think we can either use `rustls` in the example or just not offer a TLS option.

Reply:
We can use tide-rustls, we need a TLS option for some development / testing use cases. :)

Reply:
Do you want me to generate a test cert?

Reply:
That's what the related openssl code does, it generates a self signed localhost cert on the fly

Reply:
Right, but I think `tide-rustls` will only accept the cert and key via a file path. Would you prefer I make a static cert and key or use the on the fly strategy and serialize to a temp directory?

Reply:
Probably a much bigger task than you bargained for - You can use `.config()` with a rustls `ServerConfig`, and you can feed that your key/cert with https://docs.rs/rustls/0.19.0/rustls/internal/pemfile/fn.rsa_private_keys.html. BufRead is implemented by &[u8], so that means you should be able to take the generated cert as a string, do as_bytes() and then feed that to this function to provide the private key and certs in memory. 

You did way you wanted to be challenged, right ;)

Reply:
This doesn't seem too bad—mostly like just reimplementing tide-rustls. Before I begin such an undertaking though, I want to raise two possible alternatives

1. PR tide-rustls to accept certs and keys in memory
2. Use something like [tempfile](https://docs.rs/tempfile/3.1.0/tempfile/) to create temporary certs and keys.

Reply:
I think using OpenSSL's [private_key_to_der](https://docs.rs/openssl/0.10.30/openssl/pkey/struct.PKey.html#method.private_key_to_der) might work as well.

Reply:
Done—not so bad